google/cloud/spanner_dbapi/connection.py

```diff
@@ -349,6 +349,24 @@ def run_statement(self, statement, retried=False):
             ResultsChecksum() if retried else statement.checksum,
         )
 
+    def validate(self):
+        """
+        Execute a minimal request to check if the connection
+        is valid and the related database is reachable.
+
+        Raise an exception in case if the connection is closed,
+        invalid, target database is not found, or the request result
+        is incorrect.
+
+        :raises: :class:`InterfaceError`: if this connection is closed.
+        :raises: :class:`OperationalError`: if the request result is incorrect.
+        """
+        self._raise_if_closed()
+
+        with self.database.snapshot() as snapshot:
+            if [[1]] != list(snapshot.execute_sql("SELECT 1")):
+                raise OperationalError("The connection is invalid")
```

Comment:
This error message could be more helpful; currently there is no indication of why the connection is invalid.